src/main/java/org/sagebionetworks/web/client/presenter/TeamPresenter.java

```diff
@@ -85,13 +121,35 @@ public void refresh(final String teamId) {
 		synapseClient.getTeamBundle(authenticationController.getCurrentUserPrincipalId(), teamId, authenticationController.isLoggedIn(), new AsyncCallback<TeamBundle>() {
 			@Override
 			public void onSuccess(TeamBundle result) {
+				view.clear();
 				team = result.getTeam();
-				if (result.getTeamMembershipStatus() != null)
-					teamMembershipStatus = result.getTeamMembershipStatus();
-				else
-					teamMembershipStatus = null;
+				TeamMembershipStatus teamMembershipStatus = result.getTeamMembershipStatus();
 				boolean isAdmin = result.isUserAdmin();
-				view.configure(team, isAdmin, teamMembershipStatus, result.getTotalMemberCount());
+				Callback refreshCallback = new Callback() {
+					@Override
+					public void invoke() {
+						refresh(teamId);
+					}
+				};
+				view.setPublicJoinVisible(team.getCanPublicJoin());
+				view.setTotalMemberCount(result.getTotalMemberCount().toString());
+				view.setMediaObjectPanel(team);			
+				memberListWidget.configure(teamId, isAdmin, refreshCallback);				
+
+				if (teamMembershipStatus != null) {
+					if (!teamMembershipStatus.getIsMember())
+						//not a member, add Join widget
+						joinTeamWidget.configure(teamId, false, teamMembershipStatus,
+								refreshCallback, null, null, null, null, false);
+					else {
+						view.showMemberMenuItems();
+						if (isAdmin) {
+							openMembershipRequestsWidget.configure(teamId, refreshCallback);

```

Comment:
I haven't been able to get into a bad spot after making these changes, hopefully whoever ends up validating agrees.